### PR TITLE
AAP-18707 Updated procedure with EDA max running activations

### PR DESCRIPTION
**Pull request by jself-sudoku**
Updates to address issue AAP-18707 include the following:

- Deploying AAP Operator on OCP guide

     - [Chapter 6](https://access.redhat.com/documentation/en-us/red_hat_ansible_automation_platform/2.4/html/deploying_the_red_hat_ansible_automation_platform_operator_on_openshift_container_platform/deploy-eda-controller-on-aap-operator-ocp) in the [Deploying AAP operator on OCP guide](https://access.redhat.com/documentation/en-us/red_hat_ansible_automation_platform/2.4/html/deploying_the_red_hat_ansible_automation_platform_operator_on_openshift_container_platform/index).

- AAP Installation Guide
     -  [Chapter 2.4 Event-Driven Ansible controller system requirements](https://access.redhat.com/documentation/en-us/red_hat_ansible_automation_platform/2.4/html-single/red_hat_ansible_automation_platform_installation_guide/index#event-driven-ansible-system-requirements)
     - [Chapter 3.2.1.5 Single automation controller, single automation hub, and single Event-Driven Ansible controller node with external (installer managed ) database](https://access.redhat.com/documentation/en-us/red_hat_ansible_automation_platform/2.4/html-single/red_hat_ansible_automation_platform_installation_guide/index#ref-single-controller-hub-eda-with-managed-db)
     - [Appendix A.5. Event-Driven Ansible controller variables](https://access.redhat.com/documentation/en-us/red_hat_ansible_automation_platform/2.4/html-single/red_hat_ansible_automation_platform_installation_guide/index#event-driven-ansible-controller)

### CODE REVIEW

File: downstream/modules/platform/proc-deploy-eda-controller-with-aap-operator-ocp.adoc [hunk 1]
```diff
@@ -16,7 +16,24 @@
 
 . Under *Provided APIs*, locate the {EDAName} modal and click *Create instance*. 
 +
-This takes you to the Form View to customize your installation.
+This takes you to the Form View to customize your installation. 
++
+[IMPORTANT]
+====
+To ensure that you can run concurrent {EDAName} activations efficiently, you must set your maximum number of activations to 12 in the YAML view.
```
Comment (rooftopcellist) [inline downstream/modules/platform/proc-deploy-eda-controller-with-aap-operator-ocp.adoc:23]: The default value if the user does not specify anything is 12.  The wording here seems to imply that if they don't set it they are going to be in trouble.  Could we reword this to say something like:
```suggestion
To ensure that you can run concurrent {EDAName} activations efficiently, set your maximum number of activations proportional to the resources available on your cluster. You can do this by the {EDAName} settings in the YAML view. 
```

Do we have an idea of roughly how much memory/RAM each actively running rulebook activation uses?  It would be great to say something to the effect of:

> Each rulebook activation uses roughly 200Mi of memory. So if you reserve 2Gi of memory for your rulebook activations, you can expect to be about to run 8 concurrent rulebook activations. These are rough approximations for planning purposes, keep in mind that this will depend on your hardware and the tasks in your rulebook. 

These are questions we should ask Ben, Alejandro, and Elijah in the #eda-technical-discussion channel.

Comment (Alex-Izquierdo) [reply downstream/modules/platform/proc-deploy-eda-controller-with-aap-operator-ocp.adoc:23]: @rooftopcellist is right. I think there is a misunderstanding because the RHEL installer applies a default value of 12. That is the number recomended by me based on the hardware recommended for VM's. 

IMO, the goal of the documentation is to advise that number might be changed and how to do it. 
I agree that we should provide some guideline about how to scale the number of activations. 
I propose the following text:

```
When activating a rulebook under standard conditions, it typically utilizes approximately 250MB of memory. However, the actual memory consumption can vary significantly based on the complexity of your rules and the volume and size of the events processed. In scenarios where a large number of events are anticipated or the rulebook complexity is high, we strongly advise conducting a preliminary assessment of resource usage in a staging environment. This step will enable you to accurately determine the necessary number of rulebook activations. 
```

Comment (jself-sudoku) [reply downstream/modules/platform/proc-deploy-eda-controller-with-aap-operator-ocp.adoc:23]: Thanks, @Alex-Izquierdo and @rooftopcellist. I will update the content according to your suggestions and submit the PR again.

**Comment by jself-sudoku** [comment]
I have added the paragraph suggested by @Alex-Izquierdo with a few tweaks that are necessary due to IBM and RH style guide stipulations. @Alex-Izquierdo and @rooftopcellist - please review the last statement beginning with "This ensures that your maximum number..." I had to remove the "we recommend" from the statement, per style guide rules. (Notes state the following: "In most cases, avoid "recommending" that users do something, especially if the recommendation is vague or unexplained. Instead, describe why the action is problematic or beneficial."

File: downstream/modules/platform/proc-deploy-eda-controller-with-aap-operator-ocp.adoc [hunk 2]
```diff
@@ -16,7 +16,26 @@
...
-This takes you to the Form View to customize your installation.
+This takes you to the Form View to customize your installation. 
++
+[IMPORTANT]
+====
+To ensure that you can run concurrent {EDAName} activations efficiently, you must set your maximum number of activations in proportion to the resources available on your cluster. You can do this by adjusting your {EDAName} settings in the YAML view. 
+
+When activating a rulebook under standard conditions, it typically utilizes approximately 250 MB of memory. However, the actual memory consumption can vary significantly based on the complexity of your rules and the volume and size of the events processed. In scenarios where a large number of events are anticipated or the rulebook complexity is high, conduct a preliminary assessment of resource usage in a staging environment. This ensures that your maximum number of activations is based on the capacity of your resources. 
+====
++
```
Comment (jself-sudoku) [inline downstream/modules/platform/proc-deploy-eda-controller-with-aap-operator-ocp.adoc:27]: I have added the paragraph suggested by @Alex-Izquierdo with a few tweaks that are necessary due to IBM and RH style guide stipulations. @Alex-Izquierdo and @rooftopcellist - please review the last statement beginning with "This ensures that your maximum number..." I had to remove the "we recommend" from the statement, per style guide rules. (Notes state the following: "In most cases, avoid "recommending" that users do something, especially if the recommendation is vague or unexplained. Instead, describe why the action is problematic or beneficial."

File: downstream/modules/platform/ref-single-controller-hub-eda-with-managed-db.adoc [hunk 3]
```diff
@@ -101,4 +104,61 @@ sso_keystore_password=''
...
+automationedacontroller_gw_settings:
+  ANSIBLE_BASE_JWT_VALIDATE_CERT: true
+  ANSIBLE_BASE_JWT_REDIRECT_TYPE: 'eda'
+  REST_FRAMEWORK:
+    DEFAULT_AUTHENTICATION_CLASSES:
+      - 'rest_framework.authentication.SessionAuthentication'
+      - 'ansible_base.jwt_consumer.eda.auth.EDAJWTAuthentication'
+      - 'rest_framework.authentication.BasicAuthentication'
+
+## Configurable parameters
```
Comment (jself-sudoku) [inline downstream/modules/platform/ref-single-controller-hub-eda-with-managed-db.adoc:140]: @Alex-Izquierdo - I added the code from https://github.com/ansible/automation-platform-collection/blob/devel/roles/automationedacontroller/defaults/main.yml#L38 to the existing code sample in this section. Is that correct?

File: downstream/modules/platform/ref-single-controller-hub-eda-with-managed-db.adoc [hunk 4]
```diff
@@ -101,4 +104,61 @@ sso_keystore_password=''
 # automationedacontroller_ssl_cert=/path/to/automationeda.crt
 # automationedacontroller_ssl_key=/path/to/automationeda.key
 
+## Operational parameters
```
Comment (jself-sudoku) [inline downstream/modules/platform/ref-single-controller-hub-eda-with-managed-db.adoc:107]: @Alex-Izquierdo - I added the code from https://github.com/ansible/automation-platform-collection/blob/devel/roles/automationedacontroller/defaults/main.yml#L38 to the existing code sample in this section. Is that correct?

**Comment by jself-sudoku** [comment]
@Alex-Izquierdo - please let me know your thoughts about the last two RHEL updates in the AAP Install guide so that I can update it as needed.

File: downstream/modules/platform/ref-eda-system-requirements.adoc [hunk 5]
```diff
@@ -2,7 +2,13 @@
 
 = {EDAcontroller} system requirements
 
-The {EDAcontroller} is a single-node system capable of handling a variable number of long-running processes (such as, Rulebook activations) on-demand, depending on the number of CPU cores. Use the following minimum requirements to execute a maximum of 9 simultaneous activations:
+The {EDAcontroller} is a single-node system capable of handling a variable number of long-running processes (such as, Rulebook activations) on-demand, depending on the number of CPU cores. Use the following minimum requirements to execute a maximum of 12 simultaneous activations:
```
Comment (Alex-Izquierdo) [inline downstream/modules/platform/ref-eda-system-requirements.adoc:5]: Should we say that this "12" is a default setting? 
```suggestion
The {EDAcontroller} is a single-node system capable of handling a variable number of long-running processes (such as, Rulebook activations) on-demand, depending on the number of CPU cores. Use the following minimum requirements to execute by default a maximum of 12 simultaneous activations:
```

File: downstream/modules/platform/ref-single-controller-hub-eda-with-managed-db.adoc (hunk 4 above)
Comment (Alex-Izquierdo) [reply downstream/modules/platform/ref-single-controller-hub-eda-with-managed-db.adoc:107]: The right person for that answer is @jamesmarshall24

IMO putting here the code of `defaults/main.yml` might be confusing for users because appear internal variables that the customer should not know about. 

We already have a list of the supported options in https://access.redhat.com/documentation/en-us/red_hat_ansible_automation_platform/2.4/html/red_hat_ansible_automation_platform_installation_guide/appendix-inventory-files-vars#event-driven-ansible-controller

Should not be there the place to add the new `automationedacontroller_max_running_activations` variable?

Comment (jamesmarshall24) [reply downstream/modules/platform/ref-single-controller-hub-eda-with-managed-db.adoc:107]: defaults/main.yml documents variables for us as insiders to understand the value of configurable variables. Its not meant to be user facing without an editor in the middle to ensure the language is in alignment with their standards. I am not a docs writer and I wrote these descriptions to help, but I wouldn't consider my sentences the final copy of these descriptions. These shouldn't be added to the docs as presented in this PR.

Comment (jamesmarshall24) [reply downstream/modules/platform/ref-single-controller-hub-eda-with-managed-db.adoc:107]: for this particular issue, we also shouldn't be copying and pasting the entire default variables file to the documentation. There was one variable added for this feature request, so what you want to use as a reference is just `automationedacontroller_max_running_activations`

Comment (jself-sudoku) [reply downstream/modules/platform/ref-single-controller-hub-eda-with-managed-db.adoc:107]: Thank you for your input, @jamesmarshall24 . I greatly appreciate it. Will make the corrections.

File: downstream/modules/platform/ref-single-controller-hub-eda-with-managed-db.adoc [hunk 6]
```diff
@@ -101,4 +104,6 @@ sso_keystore_password=''
 # automationedacontroller_ssl_cert=/path/to/automationeda.crt
 # automationedacontroller_ssl_key=/path/to/automationeda.key
 
+automationedacontroller_max_running_activations: 12       # The number of running activations per worker. The default is set to 12.
```
Comment (jself-sudoku) [inline downstream/modules/platform/ref-single-controller-hub-eda-with-managed-db.adoc:107]: Per your recommendation, @jamesmarshall24 , all of the superfluous code info from before has been removed. Only the `automationedacontroller_max_running_activations` has been added now. Is this correct?

Comment (jself-sudoku) [reply downstream/modules/platform/ref-single-controller-hub-eda-with-managed-db.adoc:107]: Removing this line as it is not necessary since it is set to 12 as a default, per @jamesmarshall24. Thanks!

File: downstream/modules/platform/ref-eda-system-requirements.adoc (hunk 5 above)
Comment (Alex-Izquierdo) [reply downstream/modules/platform/ref-eda-system-requirements.adoc:5]: I see you have mentioned it in the other place.

File: downstream/modules/platform/ref-single-controller-hub-eda-with-managed-db.adoc (hunk 6 above)
Comment (Alex-Izquierdo) [reply downstream/modules/platform/ref-single-controller-hub-eda-with-managed-db.adoc:107]: Regardless of this line, should not be updated the list of variables in https://access.redhat.com/documentation/en-us/red_hat_ansible_automation_platform/2.4/html/red_hat_ansible_automation_platform_installation_guide/appendix-inventory-files-vars#event-driven-ansible-controller ?

File: downstream/modules/platform/ref-eda-controller-variables.adoc [hunk 7]
```diff
@@ -40,6 +40,9 @@ Default = empty list
 | *`automationedacontroller_gunicorn_workers`* | Number of workers for the API served through gunicorn.
 
 Default = (# of cores or threads) * 2 + 1
+| *`automationedacontroller_max_running_activations`* | The number of running activations per worker.
+
+Default = 12
```
Comment (jself-sudoku) [inline downstream/modules/platform/ref-eda-controller-variables.adoc:45]: @jamesmarshall24 - the `automationedacontroller_max_running_activations` variable has been added to Appendix A.5. in the install guide.

**Review by jamesmarshall24 — APPROVED** [review]

File: downstream/modules/platform/ref-eda-controller-variables.adoc [hunk 8]
```diff
@@ -40,6 +40,9 @@ Default = empty list
 | *`automationedacontroller_gunicorn_workers`* | Number of workers for the API served through gunicorn.
 
 Default = (# of cores or threads) * 2 + 1
+| *`automationedacontroller_max_running_activations`* | The number of running activations per worker.
```
Comment (Alex-Izquierdo) [inline downstream/modules/platform/ref-eda-controller-variables.adoc:43]:
```suggestion
| *`automationedacontroller_max_running_activations`* | The number of maximum running activations per node.
```

File: downstream/modules/platform/ref-single-controller-hub-eda-with-managed-db.adoc (hunk 6 above)
Comment (jself-sudoku) [reply downstream/modules/platform/ref-single-controller-hub-eda-with-managed-db.adoc]: @Alex-Izquierdo - I mentioned the update to Appendix A. 5. in the Installation Guide in my discussions with @jamesmarshall24 in our Slack discussions. On March 7, I added a link to appendix in the Conversation tab above so that this update is included in our list of changes. @jamesmarshall24 viewed and approved the update in our Slack discussion as well as the PR (attached to this reply). 

Screenshot from 2024-03-11 09-11-43

Screenshot from 2024-03-11 09-08-07

**Review by Alex-Izquierdo — APPROVED** [review]

**Comment by jself-sudoku** [comment]
@rooftopcellist - can you provide approval of the section you reviewed previously? Thank you in advance.

File: downstream/modules/platform/ref-eda-system-requirements.adoc [hunk 9]
```diff
@@ -2,7 +2,13 @@
 
 = {EDAcontroller} system requirements
 
-The {EDAcontroller} is a single-node system capable of handling a variable number of long-running processes (such as, Rulebook activations) on-demand, depending on the number of CPU cores. Use the following minimum requirements to execute a maximum of 9 simultaneous activations:
+The {EDAcontroller} is a single-node system capable of handling a variable number of long-running processes (such as, Rulebook activations) on-demand, depending on the number of CPU cores. Use the following minimum requirements to execute, by default, a maximum of 12 simultaneous activations:
```
Comment (hherbly) [inline downstream/modules/platform/ref-eda-system-requirements.adoc:5]: Punctuation suggestion: remove the comma splice after "such as" -- no comma is needed there.

Comment (hherbly) [inline downstream/modules/platform/ref-eda-system-requirements.adoc:5]: For better readability, I would move the two admonitions ("Important" and "Note") to after the minimum requirements table. You have a colon at the end of the sentence, which indicates that the minimum requirements you referenced should come next.

**Review by hherbly — APPROVED** [review]
Left a couple of small comments, but otherwise LGTM.

File: downstream/modules/platform/ref-eda-system-requirements.adoc (hunk 9 above)
Comment (jself-sudoku) [reply downstream/modules/platform/ref-eda-system-requirements.adoc:5]: good point. Pretty simple update to handle.

Comment (jself-sudoku) [reply downstream/modules/platform/ref-eda-system-requirements.adoc:5]: I considered that, but thought notes primarily came before the relevant content. As for the arrangement of the admonitions themselves, I'll follow the example in the Deploying AAP operator on OCP Guide in Chapter 6 (bullet list under "IMPORTANT"). Will send back shortly.
